### PR TITLE
Fix camera inertia problem with mouse movement

### PR DESCRIPTION
**Pull request by hexagonhexagon**
Currently when moving the camera with the mouse, there is an annoying feature that if you hold the camera still and then let go, there is some stored velocity that gets applied after you let go instead of it just staying still where you let go. I fix this behavior by adding some code to `internalUpdatePanning()` that checks if `lastMovingPosition` has been constant for the last 2 ticks: if it has, then zero out `touchPostMoveVelocity`. 

Note: I have not tested this on touch devices, and that should probably be done before pulling this.

### CODE REVIEW

**Comment by tobspr** [comment]
Uh can't we find a more elegant solution? :P

**Comment by hexagonhexagon** [comment]
Would you mind clarifying what feels so "hacky" about this? I'm trying to take a serious look at this to see how I could do it differently, but I'm not seeing what I could improve in order to do this.

File: src/js/game/camera.js [hunk 1]
```diff
@@ -712,6 +717,8 @@ export class Camera extends BasicSerializableObject {
             this.currentlyMoving = false;
             this.currentlyPinching = false;
             this.lastMovingPosition = null;
+            this.lastMovingPositionLastTick = null;
+            this.numTicksStandingStill = null;
```
Comment (tobspr) [inline src/js/game/camera.js:721]: I think we should make it consistent here and set it to `0` instead?

File: src/js/game/camera.js [hunk 2]
```diff
@@ -809,6 +816,20 @@ export class Camera extends BasicSerializableObject {
 
         this.touchPostMoveVelocity = this.touchPostMoveVelocity.multiplyScalar(velocityFade);
 
+        // Check if the camera is being dragged but standing still: if not, zero out `touchPostMoveVelocity`.
+        if (this.currentlyMoving && this.desiredCenter === null) {
+            if (this.lastMovingPositionLastTick === this.lastMovingPosition) {
+                this.numTicksStandingStill += 1;
```
Comment (tobspr) [inline src/js/game/camera.js:822]: could use `++` here

File: src/js/game/camera.js [hunk 3]
```diff
@@ -809,6 +816,20 @@ export class Camera extends BasicSerializableObject {
 
         this.touchPostMoveVelocity = this.touchPostMoveVelocity.multiplyScalar(velocityFade);
 
+        // Check if the camera is being dragged but standing still: if not, zero out `touchPostMoveVelocity`.
+        if (this.currentlyMoving && this.desiredCenter === null) {
+            if (this.lastMovingPositionLastTick === this.lastMovingPosition) {
```
Comment (tobspr) [inline src/js/game/camera.js:821]: Better use `.equalsEpsilon` here

File: src/js/game/camera.js [hunk 4]
```diff
@@ -809,6 +816,20 @@ export class Camera extends BasicSerializableObject {
...
+                this.numTicksStandingStill += 1;
+            } else {
+                this.numTicksStandingStill = 0;
+            }
+            this.lastMovingPositionLastTick = this.lastMovingPosition;
```
Comment (tobspr) [inline src/js/game/camera.js:826]: Could you use `.copy()` here? Otherwise you will share the same vector reference

**Comment by tobspr** [comment]
I guess its fine, made some annotations tho

**Comment by hexagonhexagon** [comment]
The changes you requested have been made.

**Comment by tobspr** [comment]
Nice, thank you!

**Pull request merged**